packages/react-storage/src/components/StorageBrowser/views/LocationsView/useLocationsView.ts

```diff
@@ -0,0 +1,96 @@
+import { LocationAccess } from '../../context/types';
+import { useLocationsData } from '../../context/actions';
+import { useControl } from '../../context/control';
+import React from 'react';
+import { usePaginate } from '../hooks/usePaginate';
+
+interface UseLocationsView {
+  hasNextPage: boolean;
+  hasError: boolean;
+  isLoading: boolean;
+  message: string | undefined;
+  pageItems: LocationAccess[];
+  page: number;
+  onNavigate: (location: LocationAccess) => void;
+  onRefresh: () => void;
+  onPaginateNext: () => void;
+  onPaginatePrevious: () => void;
+}
+
+export type LocationsViewActionType =
+  | { type: 'REFRESH' }
+  | { type: 'RESET' }
+  | { type: 'PAGINATE_NEXT' }
+  | { type: 'PAGINATE_PREVIOUS' }
+  | { type: 'PAGINATE'; page: number }
+  // set location to be provided to LocationDetailView
+  | { type: 'SELECT_LOCATION'; location: LocationAccess }
+  | { type: 'SEARCH'; query: string; includeSubfolders?: boolean };
```

Comment:
Does LCAG allow for delimiting subfolders?

Reply:
It doesn't, good catch i'll remove it.